src/drivers/aerofc_adc/aerofc_adc.cpp

```diff
@@ -172,59 +199,52 @@ int AEROFC_ADC::probe()
 	return -EIO;
 }
 
-int
-AEROFC_ADC::ioctl(file *filp, int cmd, unsigned long arg)
+int AEROFC_ADC::ioctl(file *filp, int cmd, unsigned long arg)
 {
 	return -ENOTTY;
 }
 
-ssize_t
-AEROFC_ADC::read(file *filp, char *buffer, size_t len)
+ssize_t AEROFC_ADC::read(file *filp, char *buffer, size_t len)
 {
-	if (len > sizeof(_samples)) {
-		len = sizeof(_samples);
+	if (len < sizeof(_sample)) {
+		return -ENOSPC;
+	}
+
+	if (len > sizeof(_sample)) {
+		len = sizeof(_sample);
 	}
 
-	pthread_mutex_lock(&_samples_mutex);
-	memcpy(buffer, _samples, len);
-	pthread_mutex_unlock(&_samples_mutex);
+	pthread_mutex_lock(&_sample_mutex);
+	memcpy(buffer, &_sample, len);
+	pthread_mutex_unlock(&_sample_mutex);
 
 	return len;
 }
 
-void
-AEROFC_ADC::cycle_trampoline(void *arg)
+void AEROFC_ADC::cycle_trampoline(void *arg)
 {
 	AEROFC_ADC *dev = reinterpret_cast<AEROFC_ADC * >(arg);
 	dev->cycle();
 }
 
-void
-AEROFC_ADC::cycle()
+void AEROFC_ADC::cycle()
 {
-	int32_t value[MAX_CHANNEL];
-
-	for (uint8_t i = 0; i < MAX_CHANNEL; i++) {
-		int ret;
-		uint8_t buffer[2];
-
-		set_address(SLAVE_ADDR | (ADC_CHANNEL_1_REG + i));
-		ret = transfer(0, 0, buffer, sizeof(buffer));
+	uint8_t buffer[2];
+	int ret;
 
-		if (ret != PX4_OK) {
-			warn("Error reading channel %u", i + 1);
-		}
+	buffer[0] = ADC_CHANNEL_REG;
+	ret = transfer(buffer, 1, buffer, sizeof(buffer));
 
-		value[i] = (int32_t)(buffer[0] | (buffer[1] << 8));
+	if (ret != PX4_OK) {
+		warn("Error reading sample");
+		return;
 	}
 
-	pthread_mutex_lock(&_samples_mutex);
+	pthread_mutex_lock(&_sample_mutex);
 
-	for (uint8_t i = 0; i < MAX_CHANNEL; i++) {
-		_samples[i].am_data = value[i];
-	}
+	_sample.am_data = (buffer[0] | (buffer[1] << 8));
```

Comment:
I mean if some in future would use the AeroFC FPGA code in another board that have a big endian processor, this bytes should be swapped, but this in unlikely to happen so I think it is ok.